var/spack/repos/builtin/packages/mapl/package.py

```diff
@@ -277,6 +321,31 @@ def cmake_args(self):
         if fflags:
             args.append(self.define("CMAKE_Fortran_FLAGS", " ".join(fflags)))
 
+        # Scripts often need to know the MPI stack used to setup the environment.
+        # Normally, we can autodetect this, but building with Spack does not
+        # seem to work. We need to pass in the MPI stack used to CMake
+        # via -DMPI_STACK on the CMake command line. We use the following
+        # names for the MPI stacks:
+        #
+        # - MPICH --> mpich
+        # - Open MPI --> openmpi
+        # - Intel MPI --> intelmpi
+        # - MVAPICH --> mvapich
+        # - HPE MPT --> mpt
```

Comment:
Hmm, I guess `mapl` doesn't use the FindMPI that comes with CMake? Or at least I can't see where `MPI_STACK` is defined / documented.

Reply:
For reasons I still don't quite understand yet, the autodetection I tried to set up in that CMake code doesn't seem to work under Spack, so our cmake gets a bit confused. 

I suppose I could change:

```cmake
  else()
    message (FATAL_ERROR "ERROR: MPI_STACK autodetection failed. Must specify a value for MPI_STACK with cmake ... -DMPI_STACK=<mpistack>. The acceptable values are: intelmpi, mvapich, mpt, mpich, openmpi")
```
to:

```cmake
  else()
    message (WARNING "WARNING: MPI_STACK autodetection failed. We will assume MPI_STACK=openmpi, but if you know you are not using it, please specify with -DMPI_STACK=mpistack. The acceptable values are: intelmpi, mvapich, mpt, mpich, openmpi")
```

so that for MAPL this is sort of unneeded. But eventually we might want to add GEOSgcm to spack and since I don't know why the autodetect code works under Spack, I'd need to add a chunk of code to the `package.py` to set MPI_STACK...

Reply:
Ok, I clearly misinterpreted the previous comments as "for some reason FindMPI doesn't work for this package". Thanks for clarifying that.